File: 프로그래머스/3/42898. 등굣길/등굣길.py
def solution(m, n, puddles):
    puddles = [[j,i] for [i,j] in puddles]
    dp = [[0] * (m+1) for _ in range(n+1)]
    dp[1][1] = 1;
    for i in range(1, n+1):
        for j in range(1, m+1):
            if i==1 and j==1: continue
            if [i,j] not in puddles:
                dp[i][j] = (dp[i-1][j] + dp[i][j-1]) % 1000000007
    return dp[n][m]

File: 프로그래머스/3/42898. 등굣길/등굣길.cpp
#include <bits/stdc++.h>
using namespace std;
#define MOD 1'000'000'007

int dp[102][102];

int solution(int m, int n, vector<vector<int>> puddles) {
    for(auto p:puddles) 
        dp[p[1]][p[0]] = -1;
    dp[1][1] = 1;
    for(int i=1;i<=n;i++) 
        for(int j=1;j<=m;j++) {
            if(i == 1 && j == 1) continue;
            if(dp[i][j] == -1) dp[i][j] = 0;
            else if(dp[i][j] != -1) dp[i][j] = (dp[i-1][j] + dp[i][j-1]) % MOD;
        }
    return dp[n][m];
}